### PR TITLE
[baxtereus] add moveit in baxter-interface

### PR DESCRIPTION
**Pull request by knorth55**
it succeeded in both simulation and real world. 
in simulation https://drive.google.com/open?id=0B5DV6gwLHtyJVEc4SnhxTjVzOG8
In real robot https://drive.google.com/open?id=0B5DV6gwLHtyJR0hpaDBoWVFMeE0

### CODE REVIEW

File: jsk_baxter_robot/baxtereus/baxter-interface.l [hunk 1]
```diff
@@ -9,9 +9,9 @@
 (defvar *wait-for-suction* 5000000)
 
 (defclass baxter-interface
-  :super robot-interface
+  :super robot-move-base-interface
```
Comment (k-okada) [inline jsk_baxter_robot/baxtereus/baxter-interface.l]: you do not have to use move-base-interface, that is for mobile robot

Comment (knorth55) [inline jsk_baxter_robot/baxtereus/baxter-interface.l]: thank you

**Comment by knorth55** [comment]
fixed

**Comment by k-okada** [comment]
do you need to override :angle-vector? if so we need to fix #719 too
https://github.com/jsk-ros-pkg/jsk_robot/blob/master/jsk_fetch_robot/fetcheus/fetch-interface.l#L26-L28

**Comment by knorth55** [comment]
i read fetcheus and it does as below.

overriding `:angle-vector-motion-plan` -> `angle-vector`,
and renaming
`angle-vector` -> `angle-vector-raw`
`angle-vector-sequence` -> `angle-vector-sequence-raw`

i thought we need to be very careful to do so because exisiting programs may not work.
but, in future, `angle-vector-motion-plan` should be mainstream and `angle-vector-raw` be an option.
Therefore, i want to override the method.
@pazeshun what do you think?

**Comment by pazeshun** [comment]
@knorth55
I agree with you. If we don't override, our followers may be confused.

**Comment by knorth55** [comment]
Overriding methods #721

**Comment by k-okada** [comment]
instead of changing code like
```
-  (send *ri* :angle-vector (send *baxter* :angle-vector) 10000)
 +  (send *ri* :angle-vector-raw (send *baxter* :angle-vector) 10000)
```
I think we should set default to nil for moveit feature and change
:angle-vector methods like
```
(:angle-vector (...)
  ;; for simulation mode   use :anlge-vector-raw
  (when (send self : simulation-modep)
   (return-from :angle-vector (send self :angle-vector-raw ...)))
 ;; when moveit is NOT enabeld, use :anlge-vector-raw
 (when (null (send self :set-moveit-environment)) ;; this just returns
moveit-evniroment
   (return-from :angle-vector (send self :angle-vector-raw ...)))
```




--
◉ Kei Okada


2016-11-11 21:55 GMT+09:00 Shingo Kitagawa <notifications@github.com>:

**Comment by pazeshun** [comment]
+1 to @k-okada

**Pull request merged**